news_scraper/middlewares.py

```diff
@@ -50,8 +53,12 @@ def process_request(self, request, spider: Spider):
             spider.logger.info("Ignoring Request (already in output): %s", request.url)
             raise IgnoreRequest
 
-        # sample web proxy usage
-        # request.meta["proxy"] = "http://user:pass@brd.superproxy.io:22225"
+        # if spider.settings.getbool("USE_PROXY"):
+        #   request.meta["proxy"] = "http://user:pass@brd.superproxy.io:22225"
+        if spider.name == "moneycontrol" and spider.settings.getbool("USE_PROXY"):
```

Comment:
This `spider.name == "moneycontrol"` has to be in test. Not the middleware. We are using proxy for testing a single spider. It has nothing to do with the core code.

Reply:
I included it so that I can test during scrapy crawl